modules/comment/logic/commentLogic.js

```diff
@@ -1,18 +1,39 @@
-const { Comment, Profile, Tip } = require('../../../models');
+const BigNumber = require('bignumber.js');
+const { Comment, Profile } = require('../../../models');
 const NotificationLogic = require('../../notification/logic/notificationLogic');
 const cache = require('../../cache/utils/cache');
+const TipLogic = require('../../tip/logic/tipLogic');
+const MdwLogic = require('../../aeternity/logic/mdwLogic');
 const { NOTIFICATION_TYPES } = require('../../notification/constants/notification');
 
 const CommentLogic = {
   async addItem(tipId, text, author, signature, challenge, parentId) {
     const parentComment = (typeof parentId !== 'undefined' && parentId !== '')
       ? await Comment.findOne({ where: { id: parentId } }) : null;
     if (parentComment === null && typeof parentId !== 'undefined' && parentId !== '') {
-      throw Error(`Could not find parent comment with id ${parentId}`);
+      throw new Error(`Could not find parent comment with id ${parentId}`);
     }
 
-    const relevantTip = await Tip.findOne({ where: { id: tipId } });
-    if (!relevantTip) throw Error(`Could not find tip with id ${tipId}`);
+    const relevantTip = await TipLogic.fetchTip(tipId);
+    if (!relevantTip) throw new Error(`Could not find tip with id ${tipId}`);
+    const parsedTip = relevantTip.toJSON();
```

Comment:
maybe refactor so we don't need `toJSON`

Reply:
we could move it to the tip logic, but I think as long as we use sequelize we will probably need `toJSON`

Reply:
can't we use raw as we do everywhere else? I didn't see the toJSON before